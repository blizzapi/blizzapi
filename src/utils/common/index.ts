import constants from '../../constants';
import {
  validateRegionId,
  validateRegionName,
  getRegionIdByName,
} from '../localization/regions';
import {
  ConstantKey,
  RegionIdOrName,
  RegionIdAsNumberOrString,
  RegionName,
} from '../../types';

export const getConstantByRegionId = (
  regionId: RegionIdAsNumberOrString,
  constantKey: ConstantKey,
) => {
  const regionIdAsString = typeof regionId !== 'string'
    ? regionId
    : regionId.toString();
  const isRegionIdValid = validateRegionId(regionIdAsString);

  if (!isRegionIdValid) {
    throw new RangeError(
      `${regionIdAsString} is not a valid parameter for getConstantByRegionId(${regionId}, '${constantKey}')`,
    );
  }
  return constants[constantKey][regionIdAsString];
};

/**
 * Returns constant value for given region name from specified constant object
 *
 * @param regionName Region name as two-letter string (e.g. 'us')
 * @param constantKey Constant key
 * @return Uri as a string
 */
export const getConstantByRegionName = (regionName: RegionName, constantKey: ConstantKey) => {
  const isRegionNameValid = validateRegionName(regionName);

  if (!isRegionNameValid) {
    throw new RangeError(
      `${regionName} is not a valid parameter for getConstantByRegionName(${regionName}, '${constantKey}')`,
    );
  }

  const regionId = getRegionIdByName(regionName);

  return constants[constantKey][regionId];
};

export const getConstantByRegion = (regionIdOrName: RegionIdOrName, constantKey: ConstantKey) =>
  (validateRegionId(regionIdOrName)
    ? getConstantByRegionId(regionIdOrName, constantKey)
    : getConstantByRegionName(regionIdOrName.toString(), constantKey));
