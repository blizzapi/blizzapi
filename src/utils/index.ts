export * from './localization';
export *  from './api';
export *  from './oauth';
