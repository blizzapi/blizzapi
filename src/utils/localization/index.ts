export * from "./regions";
export * from "./locales";
export * from "./defaultLocales";
export * from "./sc2Realms";
