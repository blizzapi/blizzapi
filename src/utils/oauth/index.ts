export * from "./authorizeUris";
export * from "./checkTokenUris";
export * from "./tokenUris";
