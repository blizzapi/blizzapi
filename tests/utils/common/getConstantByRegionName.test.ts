import { ConstantKeys } from '../../../@types';
import * as utils from '../../../src/lib/utils/common';
const { getConstantByRegionName } = utils;

import regionNamesJson from '../../../__testData__/regionNames.json';
import wrongRegionNamesJson from '../../../__testData__/wrongRegionNames.json';
import constantKeysJson from '../../../__testData__/constantKeys.json';

/* tslint:disable no-expression-statement */
describe('getConstantByRegionName()', () => {
  test('should be defined', () => {
    expect(getConstantByRegionName).toBeDefined();
  });

  test('should be of type "function"', () => {
    expect(typeof getConstantByRegionName).toBe('function');
  });

  (regionNamesJson as ReadonlyArray<string>).forEach(regionName =>
    (constantKeysJson as ConstantKeys).forEach(constantKey => {
      test(`should return correct value for ${regionName} and ${constantKey} as valid parameters`, () => {
        expect(getConstantByRegionName(regionName, constantKey)).toMatchSnapshot();
      });
    }),
  );

  (wrongRegionNamesJson as ReadonlyArray<string>).forEach(wrongRegionName =>
    (constantKeysJson as ConstantKeys).forEach(constantKey => {
      test(`should throw RangeError for ${wrongRegionName} and ${constantKey}`, () => {
        expect(() => getConstantByRegionName(wrongRegionName, constantKey)).toThrow(RangeError);
      });
    }),
  );
});
/* tslint:enable no-expression-statement */
