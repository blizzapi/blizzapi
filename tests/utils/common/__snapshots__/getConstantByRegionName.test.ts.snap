// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 1`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 2`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 3`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 4`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 5`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 6`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 7`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 8`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 9`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 10`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 11`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 12`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 13`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 14`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 15`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 16`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 17`] = `"https://api.blizzard.com.cn"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 18`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 19`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 20`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 21`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 22`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 23`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 24`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 25`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 26`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 27`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 28`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 29`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 30`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 31`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 32`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 33`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 34`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 35`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 36`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 37`] = `"https://api.blizzard.com.cn"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 38`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 39`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 40`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 41`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 42`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 43`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 44`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 45`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 46`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 47`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 48`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 49`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 50`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 51`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 52`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 53`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 54`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 55`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 56`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 57`] = `"https://api.blizzard.com.cn"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 58`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 59`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 60`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 61`] = `"https://us.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 62`] = `"https://us.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 63`] = `"https://us.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 64`] = `"https://us.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 65`] = `"https://eu.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 66`] = `"https://eu.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 67`] = `"https://eu.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 68`] = `"https://eu.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 69`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 70`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 71`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 72`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 73`] = `"https://kr.api.blizzard.com"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 74`] = `"https://apac.battle.net/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 75`] = `"https://apac.battle.net/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 76`] = `"https://apac.battle.net/oauth/check_token?token="`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 77`] = `"https://api.blizzard.com.cn"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 78`] = `"https://www.battlenet.com.cn/oauth/authorize"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 79`] = `"https://www.battlenet.com.cn/oauth/token"`;

exports[`getConstantByRegionName() should return correct value for \${regionName} and \${constantKey} as valid parameters 80`] = `"https://www.battlenet.com.cn/oauth/check_token?token="`;
